### PR TITLE
Update to readiness for 2018 edition

### PR DESCRIPTION
**Pull request by no111u3**

### CODE REVIEW

File: src/fmt.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 //! TODO write example of usage
 use core::fmt::{Result, Write};
 
-impl<Word, Error> Write for ::serial::Write<Word, Error=Error>
+impl<Word, Error> Write for dyn (::serial::Write<Word, Error=Error>)
```
Comment (luojia65) [inline src/fmt.rs]: Is it okay if `impl<Word, Error, T> Write for T where T: ::serial::Write<Word, Error=Error>` (take another proper name for T)?

Comment (no111u3) [reply src/fmt.rs]: It's ok in new version of compillers - because it replace old trait implements.

**Comment by Disasm** [comment]
Ping from triage: @thejpster, could you review this? I believe this `dyn` construction was introduced more than two stables ago (in fact, over a year ago).

**Comment by thejpster** [comment]
Needs a rebase after https://github.com/rust-embedded/embedded-hal/pull/167 landed.

**Comment by no111u3** [comment]
> Needs a rebase after #167 landed.

I close this pr because #167 landed - 2018 edition already applied. We don't need to update to becoming 2018 edition.

**Pull request closed without merging**